ledger-core/v2/application/preprocessor/templates/proxy.go.tpl

```diff
@@ -180,14 +183,22 @@ func (r *{{ $.ContractType }}) {{ $method.Name }}{{if $method.Immutable}}AsMutab
 		return {{ $method.ResultsWithErr }}
 	}
 
+	{{ if $method.Immutable }}
```

Comment:
you don't need this `if`. Every method has two variants. This one is 'Mutable'. If method `Xxx` is marked by author as immutable then this block generates method `XxxAsMutable`, so it's still call with `CallTolerable` and `CallDirty `.

Reply:
fixed